src/cli/java/org/commcare/util/screen/SessionUtils.java

```diff
@@ -0,0 +1,113 @@
+package org.commcare.util.screen;
+
+import org.commcare.cases.util.CaseDBUtils;
+import org.commcare.core.interfaces.UserSandbox;
+import org.commcare.core.parse.ParseUtils;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.model.User;
+import org.javarosa.core.services.PropertyManager;
+import org.javarosa.core.services.storage.IStorageIterator;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.PasswordAuthentication;
+import java.net.URL;
+
+public class SessionUtils {
+
+    public static void restoreUserToSandbox(UserSandbox sandbox,
+                                            SessionWrapper session,
+                                            CommCarePlatform platform,
+                                            String username,
+                                            final String password) {
+        String urlStateParams = "";
+
+        boolean failed = true;
+
+        boolean incremental = false;
+
+        if (sandbox.getLoggedInUser() != null) {
+            String syncToken = sandbox.getSyncToken();
+            String caseStateHash = CaseDBUtils.computeCaseDbHash(sandbox.getCaseStorage());
+
+            urlStateParams = String.format("&since=%s&state=ccsh:%s", syncToken, caseStateHash);
+            incremental = true;
+
+            System.out.println(String.format(
+                    "\nIncremental sync requested. \nSync Token: %s\nState Hash: %s",
+                    syncToken, caseStateHash));
+        }
+
+        PropertyManager propertyManager = platform.getPropertyManager();
+
+        //fetch the restore data and set credentials
+        String otaFreshRestoreUrl = propertyManager.getSingularProperty("ota-restore-url") +
+                "?version=2.0";
+
+        String otaSyncUrl = otaFreshRestoreUrl + urlStateParams;
+
+        String domain = propertyManager.getSingularProperty("cc_user_domain");
+        final String qualifiedUsername = username + "@" + domain;
+
+        Authenticator.setDefault(new Authenticator() {
+            @Override
+            protected PasswordAuthentication getPasswordAuthentication() {
+                return new PasswordAuthentication(qualifiedUsername, password.toCharArray());
+            }
+        });
+
+        //Go get our sandbox!
+        try {
+            System.out.println("GET: " + otaSyncUrl);
+            URL url = new URL(otaSyncUrl);
+            HttpURLConnection conn = (HttpURLConnection)url.openConnection();
+
+            if (conn.getResponseCode() == 412) {
+                System.out.println("Server Response 412 - The user sandbox is not consistent with " +
```

Comment:
should these sysouts get migrated to printStream.outs?

Reply:
:+1: thanks